components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -34,21 +34,21 @@ export default function EditPostTimeLimitModal(props: Props) {
 
     const [saving, setSaving] = useState(false);
     const [errorMessage, setErrorMessage] = useState('');
-    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || -1);
-    const [alwaysAllowPostEditing, setAlwaysAllowPostEditing] = useState(postEditTimeLimit > 0);
+    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || Constants.UNSET_POST_EDIT_TIME_LIMIT);
```

Comment:
we still should `parseInt` from the original value when we initiate the state, for one reason.
When (by any means) the value stored is anything else than a stringified number this will break, since a string with a positive length is truthy regardless of its content. In the other hand `NaN` is falsy by definition.

```suggestion
    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(parseInt(ServiceSettings?.PostEditTimeLimit, 10) || Constants.UNSET_POST_EDIT_TIME_LIMIT);
```

Reply:
For `parseInt`, TypeScript is complaining because type of `PostEditTimeLimit` is `number` or possibly `undefined` and `parseInt` is expecting a string. (`Argument of type 'number | undefined' is not assignable to parameter of type 'string'.`)
This was the reason that I was using `toString()` in the original version of my PR which you were commenting [here](https://github.com/mattermost/mattermost-webapp/pull/9979/commits/b9dc1e6789662b49e05a09338dd489c6f7279bd5#r822409698)

What would be the best way to manage this?

Reply:
Oh, I thought `PostEditTimeLimit` is of type `string`, but apparently it is a number ... this just makes it easier for you to get in the correct type:

```suggestion
    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || Constants.UNSET_POST_EDIT_TIME_LIMIT);
```